### PR TITLE
VideoPlayer: fixes

### PR DESCRIPTION
**Pull request by FernetMenta**
VideoPlayer fixes an improvements

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp [hunk 1]
```diff
@@ -1398,7 +1398,14 @@ bool CRenderManager::GetStats(int &lateframes, double &pts, int &queued, int &di
 
 void CRenderManager::CheckEnableClockSync()
 {
-  if (fabs(m_fps - g_graphicsContext.GetFPS()) < 0.01)
+  // refresh rate can be a multiple of video fps
+  double diff;
+  if (g_graphicsContext.GetFPS() >= m_fps)
+    diff = fmod(g_graphicsContext.GetFPS(), m_fps);
```
Comment (popcornmix) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: This doesn't look quite right. Suppose:
``g_graphicsContext.GetFPS() = 59.94``  ``m_fps=30``
``diff = .998`` so clocksync is not enabled, but I think it should be.

Comment (fritsch) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: fmod(59.94, 30.0) =  29.94

An additional condition if wanted could be || abs(m_fps - diff) < threshold

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: Does this work in all cases?
```
int ratio = g_graphicsContext.GetFPS() / m_fps + 0.5;
if (fabs(m_fps * ratio - g_graphicsContext.GetFPS()) < 0.01)
```
May need to protect against m_fps == 0.0.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: this is fine adjustment for a/v sync
you can't calculate an offset for pts/vblank, if fps does not exactly match refresh rate.
video pts relates to the time a frame is displayed (vblank) and ideally audio pts is synced exactly to this time.

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: Okay - I was assuming you wanted the 1001/1000 variants to be enabled - thinking the 0.01 was a 1% match, but it's actually more like a 0.01/30 = 0.03% match which excludes the 1001/1000 variants.
In that case the code does work for my example, so no problem.

**Comment by FernetMenta** [comment]
added check for m_fps == 0
jenkins build this please

**Pull request merged**